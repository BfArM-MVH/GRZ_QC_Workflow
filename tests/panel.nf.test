nextflow_pipeline {

    name "Test Workflow main.nf"
    script "../main.nf"
    config "../conf/tests/test_panel.config"

    test("Test case scenario: Panel, tumor-only. 1 samples tested with --submussion_basepath=panel --fasta=mini.fa.gz target=mini.genes.bed") {

        when {
            params {
                outdir = "${outputDir}"
            }
        }

        then {
            def stable_path = getAllFilesFromDir(params.outdir, relative:true, ignoreFile: 'tests/.nftignore')
            def report = path("${params.outdir}/report.csv")

            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // All stable path name, with a relative path
                    stable_path,
                    // All files with stable contents
                    report
                ).match() }
            )
        }

    }

}
