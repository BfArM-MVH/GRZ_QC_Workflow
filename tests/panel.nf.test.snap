{
    "Test case scenario: Panel, tumor-only. 1 samples tested with --submussion_basepath=panel --fasta=mini.fa.gz target=mini.genes.bed": {
        "content": [
            18,
            [
                "grzqc_samplesheet.csv",
                "index0_somatic0/alignments/index0_somatic0.md.bam",
                "index0_somatic0/compare_threshold/index0_somatic0.result.csv",
                "index0_somatic0/mosdepth/index0_somatic0.mosdepth.summary.txt",
                "index0_somatic0/mosdepth/index0_somatic0.per-base.bed.gz",
                "index0_somatic0/mosdepth/index0_somatic0.regions.bed.gz",
                "multiqc/multiqc_data/BETA-multiqc.parquet",
                "multiqc/multiqc_data/fastp-insert-size-plot.txt",
                "multiqc/multiqc_data/fastp-seq-content-gc-plot_Read_1_After_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-gc-plot_Read_1_Before_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-gc-plot_Read_2_After_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-gc-plot_Read_2_Before_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-n-plot_Read_1_After_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-n-plot_Read_1_Before_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-n-plot_Read_2_After_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-n-plot_Read_2_Before_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-quality-plot_Read_1_After_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-quality-plot_Read_1_Before_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-quality-plot_Read_2_After_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-quality-plot_Read_2_Before_filtering.txt",
                "multiqc/multiqc_data/fastp_filtered_reads_plot.txt",
                "multiqc/multiqc_data/fastqc-status-check-heatmap.txt",
                "multiqc/multiqc_data/fastqc_adapter_content_plot.txt",
                "multiqc/multiqc_data/fastqc_per_base_n_content_plot.txt",
                "multiqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_counts_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt",
                "multiqc/multiqc_data/mosdepth-coverage-per-contig-multi.txt",
                "multiqc/multiqc_data/mosdepth-cumcoverage-dist-id.txt",
                "multiqc/multiqc_data/mosdepth_cov_dist.txt",
                "multiqc/multiqc_data/mosdepth_cumcov_dist.txt",
                "multiqc/multiqc_data/mosdepth_perchrom.txt",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_fastp.txt",
                "multiqc/multiqc_data/multiqc_fastqc.txt",
                "multiqc/multiqc_data/multiqc_general_stats.txt",
                "multiqc/multiqc_data/multiqc_grz_qc.txt",
                "multiqc/multiqc_data/multiqc_samtools_flagstat.txt",
                "multiqc/multiqc_data/multiqc_samtools_stats.txt",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_data/samtools-flagstat-pct-table.txt",
                "multiqc/multiqc_data/samtools-flagstat-table.txt",
                "multiqc/multiqc_data/samtools-stats-dp.txt",
                "multiqc/multiqc_data/samtools_alignment_plot.txt",
                "multiqc/multiqc_plots/pdf/fastp-insert-size-plot.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-gc-plot_Read_1_After_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-gc-plot_Read_1_Before_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-gc-plot_Read_2_After_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-gc-plot_Read_2_Before_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-n-plot_Read_1_After_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-n-plot_Read_1_Before_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-n-plot_Read_2_After_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-n-plot_Read_2_Before_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-quality-plot_Read_1_After_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-quality-plot_Read_1_Before_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-quality-plot_Read_2_After_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-quality-plot_Read_2_Before_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp_filtered_reads_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/fastp_filtered_reads_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-status-check-heatmap.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_adapter_content_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_base_n_content_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_base_sequence_quality_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Counts.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Percentages.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_quality_scores_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_counts_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_counts_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_duplication_levels_plot.pdf",
                "multiqc/multiqc_plots/pdf/grz_qc.pdf",
                "multiqc/multiqc_plots/pdf/mosdepth-coverage-per-contig-multi-cnt.pdf",
                "multiqc/multiqc_plots/pdf/mosdepth-coverage-per-contig-multi-log.pdf",
                "multiqc/multiqc_plots/pdf/mosdepth-cumcoverage-dist-id.pdf",
                "multiqc/multiqc_plots/pdf/samtools-flagstat-pct-table.pdf",
                "multiqc/multiqc_plots/pdf/samtools-flagstat-table.pdf",
                "multiqc/multiqc_plots/pdf/samtools-stats-dp.pdf",
                "multiqc/multiqc_plots/pdf/samtools_alignment_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/samtools_alignment_plot-pct.pdf",
                "multiqc/multiqc_plots/png/fastp-insert-size-plot.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-gc-plot_Read_1_After_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-gc-plot_Read_1_Before_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-gc-plot_Read_2_After_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-gc-plot_Read_2_Before_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-n-plot_Read_1_After_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-n-plot_Read_1_Before_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-n-plot_Read_2_After_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-n-plot_Read_2_Before_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-quality-plot_Read_1_After_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-quality-plot_Read_1_Before_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-quality-plot_Read_2_After_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-quality-plot_Read_2_Before_filtering.png",
                "multiqc/multiqc_plots/png/fastp_filtered_reads_plot-cnt.png",
                "multiqc/multiqc_plots/png/fastp_filtered_reads_plot-pct.png",
                "multiqc/multiqc_plots/png/fastqc-status-check-heatmap.png",
                "multiqc/multiqc_plots/png/fastqc_adapter_content_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_base_n_content_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_base_sequence_quality_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Counts.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Percentages.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_quality_scores_plot.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_counts_plot-cnt.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_counts_plot-pct.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_duplication_levels_plot.png",
                "multiqc/multiqc_plots/png/grz_qc.png",
                "multiqc/multiqc_plots/png/mosdepth-coverage-per-contig-multi-cnt.png",
                "multiqc/multiqc_plots/png/mosdepth-coverage-per-contig-multi-log.png",
                "multiqc/multiqc_plots/png/mosdepth-cumcoverage-dist-id.png",
                "multiqc/multiqc_plots/png/samtools-flagstat-pct-table.png",
                "multiqc/multiqc_plots/png/samtools-flagstat-table.png",
                "multiqc/multiqc_plots/png/samtools-stats-dp.png",
                "multiqc/multiqc_plots/png/samtools_alignment_plot-cnt.png",
                "multiqc/multiqc_plots/png/samtools_alignment_plot-pct.png",
                "multiqc/multiqc_plots/svg/fastp-insert-size-plot.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-gc-plot_Read_1_After_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-gc-plot_Read_1_Before_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-gc-plot_Read_2_After_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-gc-plot_Read_2_Before_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-n-plot_Read_1_After_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-n-plot_Read_1_Before_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-n-plot_Read_2_After_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-n-plot_Read_2_Before_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-quality-plot_Read_1_After_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-quality-plot_Read_1_Before_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-quality-plot_Read_2_After_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-quality-plot_Read_2_Before_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp_filtered_reads_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/fastp_filtered_reads_plot-pct.svg",
                "multiqc/multiqc_plots/svg/fastqc-status-check-heatmap.svg",
                "multiqc/multiqc_plots/svg/fastqc_adapter_content_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_base_n_content_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_base_sequence_quality_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Counts.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Percentages.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_quality_scores_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_counts_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_counts_plot-pct.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_duplication_levels_plot.svg",
                "multiqc/multiqc_plots/svg/grz_qc.svg",
                "multiqc/multiqc_plots/svg/mosdepth-coverage-per-contig-multi-cnt.svg",
                "multiqc/multiqc_plots/svg/mosdepth-coverage-per-contig-multi-log.svg",
                "multiqc/multiqc_plots/svg/mosdepth-cumcoverage-dist-id.svg",
                "multiqc/multiqc_plots/svg/samtools-flagstat-pct-table.svg",
                "multiqc/multiqc_plots/svg/samtools-flagstat-table.svg",
                "multiqc/multiqc_plots/svg/samtools-stats-dp.svg",
                "multiqc/multiqc_plots/svg/samtools_alignment_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/samtools_alignment_plot-pct.svg",
                "report.csv",
                "report.xlsx"
            ],
            "report.csv:md5,3bd5298d860fc108d853dbfa6f562b64"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-07-21T15:26:48.713165498"
    }
}